From 1116dab4f2e03dadd2e1a427084536a2f14cfb19 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jozef=20=C4=8C=C3=AD=C5=BE?= <jozef.ciz.ml@gmail.com>
Date: Tue, 12 Apr 2022 22:35:14 +0200
Subject: [PATCH] Add more keyboard shortcuts

Add support for redo by ctrl+shift+z
Add PgUp PgDown movement navigation to standard mode
Add more zoom shortcuts (C-- and C-= to all modes).
Add common ctrl-based copy-paste shortucts to vim mode.
---
 src/mode.cpp          | 5 +++++
 src/mode_standard.cpp | 6 ++++++
 2 files changed, 11 insertions(+)

diff --git a/src/mode.cpp b/src/mode.cpp
index 6835eb78..7aa790c1 100644
--- a/src/mode.cpp
+++ b/src/mode.cpp
@@ -2662,6 +2662,11 @@ void ZepMode::AddGlobalKeyMaps()
     keymap_add({ &m_normalMap }, { "<C-i><C-o>" }, id_SwitchToAlternateFile);
     keymap_add({ &m_normalMap }, { "+" }, id_FontBigger);
     keymap_add({ &m_normalMap }, { "-" }, id_FontSmaller);
+    keymap_add({ &m_normalMap, &m_visualMap, &m_insertMap }, { "<C-=>" }, id_FontBigger);
+    keymap_add({ &m_normalMap, &m_visualMap, &m_insertMap }, { "<C-->" }, id_FontSmaller);
+    keymap_add({ &m_insertMap }, { "<C-v>" }, id_StandardPaste);
+    keymap_add({ &m_normalMap, &m_visualMap }, { "<C-S-c>" }, id_StandardCopy);
+    keymap_add({ &m_normalMap, &m_visualMap, &m_insertMap }, { "<C-S-v>" }, id_StandardPaste);
 }
 
 CursorType ZepMode::GetCursorType() const
diff --git a/src/mode_standard.cpp b/src/mode_standard.cpp
index 8fd451e1..3416e680 100644
--- a/src/mode_standard.cpp
+++ b/src/mode_standard.cpp
@@ -51,6 +51,7 @@ void ZepMode_Standard::Init()
     keymap_add({ &m_insertMap }, { "<Tab>" }, id_InsertTab);
     keymap_add({ &m_insertMap, &m_visualMap }, { "<Del>" }, id_Delete);
     keymap_add({ &m_insertMap, &m_visualMap }, { "<C-y>" }, id_Redo);
+    keymap_add({ &m_insertMap, &m_visualMap }, { "<C-S-z>" }, id_Redo);
     keymap_add({ &m_insertMap, &m_visualMap }, { "<C-z>" }, id_Undo);
 
     keymap_add({ &m_insertMap, &m_visualMap }, { "<Left>" }, id_MotionStandardLeft);
@@ -61,6 +62,8 @@ void ZepMode_Standard::Init()
     keymap_add({ &m_insertMap }, { "<Home>" }, id_MotionLineHomeToggle);
     keymap_add({ &m_insertMap }, { "<C-Left>" }, id_MotionStandardLeftWord);
     keymap_add({ &m_insertMap }, { "<C-Right>" }, id_MotionStandardRightWord);
+    keymap_add({ &m_insertMap }, { "<PageDown>" }, id_MotionPageForward);
+    keymap_add({ &m_insertMap }, { "<PageUp>" }, id_MotionPageBackward);
 
     keymap_add({ &m_insertMap, &m_visualMap }, { "<C-S-Left>" }, id_MotionStandardLeftWordSelect);
     keymap_add({ &m_insertMap, &m_visualMap }, { "<C-S-Right>" }, id_MotionStandardRightWordSelect);
@@ -80,6 +83,9 @@ void ZepMode_Standard::Init()
 
     keymap_add({ &m_normalMap, &m_visualMap, &m_insertMap }, { "<Escape>" }, id_InsertMode);
     keymap_add({ &m_normalMap }, { "<Backspace>" }, id_MotionStandardLeft);
+
+    keymap_add({ &m_normalMap, &m_visualMap, &m_insertMap }, { "<C-=>" }, id_FontBigger);
+    keymap_add({ &m_normalMap, &m_visualMap, &m_insertMap }, { "<C-->" }, id_FontSmaller);
 }
 
 void ZepMode_Standard::Begin(ZepWindow* pWindow)
-- 
2.39.1

